kettle/README.md

```diff
@@ -100,10 +100,25 @@ gs://<bucket path>: #bucket url
 
 A [postsubmit job](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/test-infra/test-infra-trusted.yaml#L203-L210) runs that pushes Kettle on changes.
 
+# PubSub
+
+Kettle `stream.py` leverages Google Cloud [PubSub] to alert on GCS changes. These events are tied to the `gcs-changes` Topic in the `kubernetes-jenkins` project where Prow job artifacts are collated. Each time an artifact is finalized, a PubSub event is triggered and Kettle collects job information when it sees a resource uploaded called `finished.json` (indicating the build completed).
```

Comment:
Is the topic something we need to create?

The bucket and project are named the same, is this getting message for all buckets in the project, or just gs://kubernetes-jenkins?

Does this work for any of the other buckets listed in buckets.yaml or is this hardcoded?

Reply:
This is hardcoded as part of the subscription. I am not 100% how the topic was created. I believe it is tied to the jobs in `kubernetes-jenkins` exclusively. We could setup a per-bucket event using the prefixes and fork the process across all buckets but would require topics and subscriptions created in each bucket + permission for kettle to access it, would be involved.

Reply:
based on https://cloud.google.com/storage/docs/reporting-changes

I think the topic may have been auto-created by this command:
```
gsutil notification create -t projects/kubernetes-jenkins/topics/gcs-changes -f json gs://kubernetes-jenkins
```

Reply:
also maybe change first sentence to "alert on GCS changes in the `kubernetes-jenkins` bucket"

Reply:
Done